chapter_natural-language-processing-applications/natural-language-inference-bert.md

```diff
@@ -313,7 +311,12 @@ class BERTClassifier(nn.Module):
         super(BERTClassifier, self).__init__()
         self.encoder = bert.encoder
         self.hidden = bert.hidden
-        self.output = nn.Linear(256, 3)
+        self.output = nn.LazyLinear(3)
+        self.encoder(torch.zeros([1,128], dtype=torch.long), 
+                     torch.zeros([1,128], dtype=torch.long),
+                     torch.zeros([1], dtype=torch.long))
+        self.hidden(torch.zeros([1,256]))
+        self.output(torch.zeros([1,256]))
```

Comment:
Thanks! Are these workarounds best practice or necessary? Is it possible to avoid hardcoding 128 and 256?

Reply:
Besides, please add space after commas

Reply:
I removed these hardcodes and ran the forward pass with train_iter for lazy initialization.

Reply:
Thanks. Resolving it now.